cmd/simulator/txs/agent.go

```diff
@@ -75,12 +76,6 @@ func (a issueNAgent[T]) Execute(ctx context.Context) error {
 		totalConfirmedTime time.Duration
 	)
 
-	defer func() {
-		if err := a.worker.Close(ctx); err != nil {
```

Comment:
where are the workers closed now?

Reply:
The workers are no longer closed. It wasn't strictly necessary to close them before, so it's now up to the creator of those clients to call `Close` if desired.